### PR TITLE
Part 1 complete

### PR DESCRIPTION
**Pull request by erikstmartin**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -92,6 +92,59 @@ good to go!
  */
 
 #include <iostream>
+
+struct FloatType
+{
+public:
+    float add(float lhs, float rhs) { return lhs + rhs; }
```
Comment (matkatmusic) [inline main.cpp]: `struct` is `public` by default.
this access specifier is unnecessary
- [x] revise

File: main.cpp [hunk 2]
```diff
@@ -92,6 +92,59 @@ good to go!
...
+        {
+            std::cout << "\nwarning, floating point division by zero returns 'inf' !" << std::endl;
+        }
+        return lhs / rhs; 
+    }
+};
+
+struct DoubleType
+{
+public:
```
Comment (matkatmusic) [inline main.cpp]: `struct` is `public` by default.
this access specifier is unnecessary
- [x] revise

File: main.cpp [hunk 3]
```diff
@@ -92,6 +92,59 @@ good to go!
...
+        {
+            std::cout << "\nwarning, floating point division by zero returns 'inf' !" << std::endl;
+        }
+        return lhs / rhs; 
+    }
+};
+
+struct IntType
+{
+public:
```
Comment (matkatmusic) [inline main.cpp]: `struct` is `public` by default.
this access specifier is unnecessary
- [x] revise

File: main.cpp [hunk 4]
```diff
@@ -92,6 +92,59 @@ good to go!
...
+    int add(int lhs, int rhs) { return lhs + rhs; }
+    int subtract(int lhs, int rhs) { return lhs - rhs; }
+    int multiply(int lhs, int rhs) { return lhs * rhs; }
+    int divide(int lhs, int rhs)
+    {
+        if (rhs == 0)
+        {
+            std::cout << "error, integer division by zero will crash the program!\nreturn lhs" << std::endl;
+            return lhs;
+        }
```
Comment (matkatmusic) [inline main.cpp]: This `std::cout` statement does not produce an identical message when compared to the expected project output using the diffchecker linked in the instructions.
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 4 above)
Comment (erikstmartin) [reply main.cpp]: I used the diffchecker, it showed it as identical. I've refactored it to two lines to make it more clear that it produces two lines of output though.

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**